packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js

```diff
@@ -0,0 +1,125 @@
+    WHERE dr.child_id = di.id AND di.id IN (${idList});
+`);
+  await db.runSql(`
+    DELETE FROM legacy_report lr
+    USING dashboard_item di
+    WHERE lr.code = di.code AND di.id IN (${idList});
+`);
+  await db.runSql(`DELETE FROM dashboard_item WHERE id IN (${idList})`);
+};
+
+const deleteMapOverlayGroups = async (db, codes) => {
+  const codeList = arrayToDbString(codes);
+
+  await db.runSql(`
+    DELETE FROM map_overlay_group_relation mogr
+    USING map_overlay_group mog
+    WHERE mogr.map_overlay_group_id = mog.id AND mog.code IN (${codeList});
+  `);
+  await db.runSql(`
+    DELETE FROM map_overlay_group_relation mogr
+    USING map_overlay_group mog
+    WHERE mogr.child_id = mog.id AND mog.code IN (${codeList});
+`);
+  await db.runSql(`DELETE FROM map_overlay_group WHERE code IN (${codeList})`);
+};
+
+const deleteMapOverlaysByIds = async (db, ids) => {
+  const idList = arrayToDbString(ids);
+
+  await db.runSql(`
+    DELETE FROM map_overlay_group_relation mogr
+    USING "mapOverlay" mo
+    WHERE mogr.child_id = mo.id AND mo.id IN (${idList});
+`);
+  await db.runSql(`DELETE FROM "mapOverlay" WHERE id IN (${idList})`);
+};
+
+const deleteProjectAndHierarchy = async (db, code) => {
+  const { rows } = await db.runSql(`SELECT * FROM project WHERE code = '${code}'`);
+  const [{ id: projectId, entity_hierarchy_id: entityHierarchyId }] = rows;
+
+  await db.runSql(`DELETE FROM access_request WHERE project_id = '${projectId}'`);
+  await db.runSql(`DELETE FROM project WHERE code = '${code}'`);
+
+  await toggleEntityTriggers(db, false);
+  await db.runSql(
+    `DELETE FROM ancestor_descendant_relation WHERE entity_hierarchy_id = '${entityHierarchyId}'`,
+  );
+  await db.runSql(`DELETE FROM entity_relation WHERE entity_hierarchy_id = '${entityHierarchyId}'`);
+  await db.runSql(`DELETE FROM entity_hierarchy WHERE id = '${entityHierarchyId}'`);
+  await db.runSql(`DELETE FROM entity WHERE code = '${code}'`);
+  await toggleEntityTriggers(db, true);
+};
+
+const toggleEntityTriggers = async (db, enable) => {
+  const action = enable ? 'ENABLE' : 'DISABLE';
+
+  await db.runSql(`ALTER TABLE entity_relation ${action} TRIGGER entity_relation_trigger`);
+  await db.runSql(`ALTER TABLE entity_hierarchy ${action} TRIGGER entity_hierarchy_trigger`);
+  await db.runSql(`ALTER TABLE entity ${action} TRIGGER entity_trigger`);
+};
+
+exports.up = async function (db) {
+  await deleteDashboards(db, IMMS_DASHBOARDS);
+  const { rows: immsDashboardItems } = await db.runSql(
+    `SELECT * FROM dashboard_item WHERE code ILIKE 'Imms_%';`,
+  );
+  const immsDashboardItemIds = immsDashboardItems.map(di => di.id);
+  await deleteDashboardItemsByIds(db, immsDashboardItemIds);
+
+  await deleteMapOverlayGroups(db, IMMS_MAP_OVERLAY_GROUPS);
+  const { rows: immsMapOverlays } = await db.runSql(`
+      SELECT * FROM "mapOverlay"
+      WHERE "projectCodes" IN ('{imms,explore}')`);
+  const immsMapOverlayIds = immsMapOverlays.map(di => di.id);
```

Comment:
I noticed there's also some other map overlays that has `imms` in their project codes, but used in other projects as well. So maybe we want to remove `imms` from their projectCodes as well?

Query:
```
SELECT * FROM "mapOverlay"
WHERE "projectCodes"::text like '%imms%'
```

Reply:
Oops good point!